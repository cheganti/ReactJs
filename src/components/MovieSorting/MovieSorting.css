.sortByDiv {
  text-align: right;
  position: relative;
  color: #ffffff;
}
