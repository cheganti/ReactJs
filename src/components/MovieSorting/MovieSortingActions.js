export const SET_SORT_BY_RATING = 'SET_SORT_BY_RATING';
export const SET_SORT_BY_RELEASEDATE = 'SET_SORT_BY_RELEASEDATE';


export const setSortByRating = () => ({ type: SET_SORT_BY_RATING });
export const setSortByReleaseDate = () => ({ type: SET_SORT_BY_RELEASEDATE });
