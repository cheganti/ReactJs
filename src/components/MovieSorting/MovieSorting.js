import React from 'react';
import PropTypes from 'prop-types';
import './MovieSorting.css';

export default function MovieSorting(props) {
  const {
    count,
    sortByRelease,
    isRating,
    sortByRating,
    isRelease,
  } = props;
  return (
    <div className="sortByDiv">
      <div className="float-left">
        <span className="bold">Count: </span>
        <span>{count}</span>
      </div>
      <div className="float-right">
        <span className="bold">Sort By:</span>
        <button
          type="button"
          onClick={sortByRelease}
          className={`btn btn-success-cus ${isRating === true
            ? 'active'
            : 'deactive'}`}
        >
          Release Date
        </button>
        <button
          type="button"
          onClick={sortByRating}
          className={`btn btn-success-cus ${isRelease === true
            ? 'active'
            : 'deactive'}`}
        >
          Rating
        </button>
      </div>
    </div>
  );
}
MovieSorting.propTypes = {
  count: PropTypes.number.isRequired,
  isRating: PropTypes.bool.isRequired,
  isRelease: PropTypes.bool.isRequired,
  sortByRelease: PropTypes.func.isRequired,
  sortByRating: PropTypes.func.isRequired,
};
