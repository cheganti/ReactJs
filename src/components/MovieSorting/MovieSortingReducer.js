import { SET_SORT_BY_RATING, SET_SORT_BY_RELEASEDATE } from './MovieSortingActions';

function MovieSortingReducer(state = {
  activeOnSort: 'release',
}, action) {
  switch (action.type) {
    case SET_SORT_BY_RATING:
      return {
        ...state,
        activeOnSort: 'rating',
      };
    case SET_SORT_BY_RELEASEDATE:
      return {
        ...state,
        activeOnSort: 'release',
      };
    default:
      return state;
  }
}
export default MovieSortingReducer;
