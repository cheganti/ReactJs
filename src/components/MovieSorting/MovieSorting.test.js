import React from 'react';
import { shallow } from 'enzyme';
import MovieSorting from './MovieSorting';

describe('MovieSorting', () => {
  it('renders without breaking', () => {
    const wrapper = shallow(<MovieSorting count="10" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('onClick SortByRelease', () => {
    const handleRelease = jest.fn();
    const wrapper = shallow(<MovieSorting sortByRelease={handleRelease} />);
    const button = wrapper.find('button').at(0);
    button.simulate('click');
    expect(handleRelease).toHaveBeenCalled();
  });
  it('onClick SortByRating', () => {
    const handleRating = jest.fn();
    const wrapper = shallow(<MovieSorting sortByRating={handleRating} />);
    const button = wrapper.find('button').at(1);
    button.simulate('click');
    expect(handleRating).toHaveBeenCalled();
  });
});
