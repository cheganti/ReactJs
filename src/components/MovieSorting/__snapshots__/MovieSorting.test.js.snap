// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSorting renders without breaking 1`] = `
<div
  className="sortByDiv"
>
  <div
    className="float-left"
  >
    <span
      className="bold"
    >
      Count: 
    </span>
    <span>
      10
    </span>
  </div>
  <div
    className="float-right"
  >
    <span
      className="bold"
    >
      Sort By:
    </span>
    <button
      className="btn btn-success-cus deactive"
      type="button"
    >
      Release Date
    </button>
    <button
      className="btn btn-success-cus deactive"
      type="button"
    >
      Rating
    </button>
  </div>
</div>
`;
